addons/l10n_it_edi/models/account_edi.py

```diff
@@ -0,0 +1,419 @@
+# -*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields, _
+from odoo.tests.common import Form
+from odoo.exceptions import UserError
+from odoo.tools import float_repr
+
+import re
+from datetime import date, datetime
+import logging
+import base64
+
+
+_logger = logging.getLogger(__name__)
+
+DEFAULT_FACTUR_ITALIAN_DATE_FORMAT = '%Y-%m-%d'
+
+
+class AccountEdi(models.Model):
+    _inherit = 'account.edi'
+
+    def _create_invoice_from_xml_tree(self, filename, tree):
+        self.ensure_one()
+        invoice = super()._create_invoice_from_xml_tree(filename, tree)
+        if invoice:
+            return invoice
+        if self.code != 'fattura_pa':
+            return invoice
+        if not re.search("([A-Z]{2}[A-Za-z0-9]{2,28}_[A-Za-z0-9]{0,5}.(xml.p7m|xml))", filename):
+            return invoice
+        else:
+            invoice = self.env['account.move'].create({})
+            return self._import_fattura_pa(tree, invoice)
+
+    def _update_invoice_from_xml_tree(self, filename, tree, invoice):
+        self.ensure_one()
+        if self.code != 'fattura_pa':
+            return super()._update_invoice_from_xml_tree(filename, tree, invoice)
+        if not re.search("([A-Z]{2}[A-Za-z0-9]{2,28}_[A-Za-z0-9]{0,5}.(xml.p7m|xml))", filename):
+            return False
+        else:
+            return self._import_fattura_pa(tree, invoice)
+
+    def _import_fattura_pa(self, tree, invoice):
+        """ Decodes a fattura_pa invoice into an invoice.
+
+        :param tree: the fattura_pa tree to decode.
+        :param invoice: the invoice to set the decoded data.
+        :returns: the created invoice.
+        """
+        invoices = self.env['account.move']
+        multi = False
+
+        # possible to have multiple invoices in the case of an invoice batch, the batch itself is repeated for every invoice of the batch
+        for body_tree in tree.xpath('//FatturaElettronicaBody'):
+            if multi or not invoice:
+                # make sure all the iterations create a new invoice record (except the first which could have already created one)
+                invoice = self.env['account.move']
+            multi = True
```

Comment:
As the multi mode is managed outside this method, the multi is simply a loot using the Form view